src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/boss_ignis.cpp

```diff
@@ -82,300 +85,306 @@ enum
+        AddCombatAction(IGNIS_BERSERK, 10min);
     }
 
-    void KilledUnit(Unit* pVictim) override
-    {
-        if (pVictim->GetTypeId() != TYPEID_PLAYER)
-            return;
-
-        DoScriptText(urand(0, 1) ? SAY_SLAY_1 : SAY_SLAY_2, m_creature);
-    }
+    instance_ulduar* m_instance;
+    bool m_isRegularMode;
 
-    void Aggro(Unit* /*pWho*/) override
+    void JustRespawned() override
     {
-        if (m_pInstance)
-            m_pInstance->SetData(TYPE_IGNIS, IN_PROGRESS);
-
-        DoScriptText(SAY_AGGRO, m_creature);
+        BossAI::JustRespawned();
+        m_creature->GetMap()->GetVariableManager().SetVariable(WORLD_STATE_CUSTOM_RESPAWN_IGNIS_ADDS, 0);
     }
 
-    void JustReachedHome() override
+    void EnterEvadeMode() override
     {
-        if (m_pInstance)
-            m_pInstance->SetData(TYPE_IGNIS, FAIL);
+        BossAI::EnterEvadeMode();
+        GuidSet constructs;
+        m_instance->GetIronConstructGuids(constructs);
+        for (const ObjectGuid& guid : constructs)
+        {
+            Creature* construct = m_creature->GetMap()->GetCreature(guid);
+            if (!construct)
+                continue;
+            construct->SetRespawnDelay(30);
+            construct->ForcedDespawn();
+        }
+        m_creature->GetMap()->GetVariableManager().SetVariable(WORLD_STATE_CUSTOM_RESPAWN_IGNIS_ADDS, 1);
     }
 
-    void JustSummoned(Creature* pSummoned) override
+    void JustSummoned(Creature* summoned) override
     {
-        if (pSummoned->GetEntry() == NPC_SCORCH)
-            pSummoned->CastSpell(pSummoned, m_bIsRegularMode ? SPELL_SCORCH_AURA : SPELL_SCORCH_AURA_H, TRIGGERED_OLD_TRIGGERED);
+        if (summoned->GetEntry() == NPC_SCORCH)
+        {
+            summoned->CastSpell(nullptr, m_isRegularMode ? SPELL_SCORCH_AURA : SPELL_SCORCH_AURA_H, TRIGGERED_OLD_TRIGGERED);
+            m_creature->AddSummonForOnDeathDespawn(summoned->GetObjectGuid());
+        }
     }
 
     // TODO: Use the vehicle boarding wrappers when they are implemented in core
-    void SpellHit(Unit* pCaster, const SpellEntry* pSpell) override
+    void SpellHit(Unit* caster, const SpellEntry* spell) override
     {
-        if (pCaster->GetTypeId() != TYPEID_PLAYER)
+        if (caster->GetTypeId() != TYPEID_PLAYER)
             return;
 
         // Handle the case when passenger is loaded to the second seat
-        if (pSpell->Id == SPELL_GRAB_POT)
-            DoCastSpellIfCan(pCaster, SPELL_SLAG_POT, CAST_TRIGGERED);
+        if (spell->Id == SPELL_GRAB_POT)
+            DoCastSpellIfCan(caster, SPELL_SLAG_POT, CAST_TRIGGERED);
     }
 
-    void UpdateAI(const uint32 uiDiff) override
+    void ExecuteAction(uint32 action) override
     {
-        if (!m_creature->SelectHostileTarget() || !m_creature->GetVictim())
-            return;
-
-        if (m_uiBerserkTimer)
+        switch (action)
         {
-            if (m_uiBerserkTimer <= uiDiff)
+            case IGNIS_BERSERK:
```

Comment:
Should also be in spell lists. I mean you can make it repeat after duration.

Reply:
I have confirmation for some bosses that they do repeat their Enrage/Berserk after its run out, but I haven't seen it for Ignis, which is why I did it like this. I'll double check and add it to spell list